models/models/output_file.py

```diff
@@ -0,0 +1,216 @@
+from pathlib import Path
+from typing import TypeAlias
+
+from cloudpathlib import AnyPath, CloudPath, GSPath
+from google.cloud.storage import Client
+from pydantic import BaseModel
+
+from models.base import SMBase, parse_sql_bool
+
+RecursiveDict: TypeAlias = dict[str, 'str | RecursiveDict']
+
+
+class OutputFileInternal(SMBase):
+    """File model for internal use"""
+
+    id: int
+    parent_id: int | None = None
+    path: str
+    basename: str
+    dirname: str
+    nameroot: str
+    nameext: str | None
+    file_checksum: str | None
+    size: int
+    meta: dict | None = None
+    valid: bool = False
+    secondary_files: dict | None = {}
+
+    @staticmethod
+    def from_db(**kwargs):
+        """
+        Convert from db keys, mainly converting id to id_
+        """
+        return OutputFileInternal(
+            id=kwargs.pop('id'),
+            parent_id=kwargs.pop('parent_id', None),
+            path=kwargs.get('path'),
+            basename=kwargs.get('basename'),
+            dirname=kwargs.get('dirname'),
+            nameroot=kwargs.get('nameroot'),
+            nameext=kwargs.get('nameext'),
+            file_checksum=kwargs.get('file_checksum'),
+            size=kwargs.get('size'),
+            meta=kwargs.get('meta'),
+            valid=parse_sql_bool(kwargs.get('valid')),
+            secondary_files=kwargs.get('secondary_files', {}),
+        )
+
+    def to_external(self):
+        """
+        Convert to external model
+        """
+        return OutputFile(
+            id=self.id,
+            parent_id=self.parent_id,
+            path=self.path,
+            basename=self.basename,
+            dirname=self.dirname,
+            nameroot=self.nameroot,
+            nameext=self.nameext,
+            file_checksum=self.file_checksum,
+            size=self.size,
+            meta=self.meta,
+            valid=self.valid,
+        )
+
+    @staticmethod
+    async def get_file_info(
+        file_obj: CloudPath | AnyPath | GSPath | Path, client: Client
+    ) -> dict | None:
+        """Get file info for file at given path"""
+        try:
+            file_checksum = None
+            valid = False
+            size = 0
+            # try:
+            if isinstance(file_obj, GSPath) and client:
+                if '/' in file_obj.blob:
+                    prefix = '/'.join(file_obj.blob.split('/')[:-1]) + '/'
+                else:
+                    prefix = ''
+                delimiter = '/'
+                blobs = client.list_blobs(
+                    bucket_or_name=str(file_obj.bucket),
+                    versions=False,
+                    prefix=prefix,
+                    delimiter=str(delimiter),
+                )  # pylint: disable=E1101
+                for blob in blobs:
+                    if blob.name == file_obj.blob:
+                        if file_obj.suffix != '.mt':
```

Comment:
Would be good to have a note here explaining why we only set checksums when it is a .mt file